pkg/agent/stream/client.go

```diff
@@ -214,3 +224,39 @@ func (s *StreamerClient) Healthcheck(_ string) *hc.Status {
 		Result: hc.OK,
 	}
 }
+
+func (s *StreamerClient) UpdateAgentStatus(state, prevState, message string) error {
+	// Initial running status and stopped status set by watch-controller
+	// allow running status after recovery from unhealthy state
+	if canTransitionAgentState(state, prevState) {
+		return s.writeStatusRequest(state, message)
+	} else {
```

Comment:
I think you can drop this else statement, since the return happens in the if block